systemtests/src/main/java/com/github/streamshub/systemtests/TestCaseConfig.java

```diff
@@ -0,0 +1,49 @@
+package com.github.streamshub.systemtests;
+
+import com.github.streamshub.systemtests.utils.PwUtils;
+import com.github.streamshub.systemtests.utils.Utils;
+import com.microsoft.playwright.Browser;
+import com.microsoft.playwright.BrowserContext;
+import com.microsoft.playwright.Page;
+import com.microsoft.playwright.Playwright;
+import org.junit.jupiter.api.extension.ExtensionContext;
+
+import java.util.Locale;
+
+
+public class TestCaseConfig {
+    private final String testName;
+    private final String namespace;
+    private final Playwright playwright;
+    private final Browser browser;
+    private final BrowserContext context;
+    private final Page page;
+
+    public TestCaseConfig(ExtensionContext extensionContext) {
+        this.testName = extensionContext.getTestMethod().isPresent() ? extensionContext.getTestMethod().get().getName() : extensionContext.getTestClass().get().getSimpleName();
+        this.namespace = extensionContext.getTestClass().isPresent() ?  extensionContext.getTestClass().get().getSimpleName().toLowerCase(Locale.ENGLISH) + "-" + Utils.hashStub(testName) : "nullClass";
```

Comment:
A few adjustments to the `Optional` handling to clean this up and make Sonar happy:

```suggestion
        this.testName = extensionContext.getTestMethod().map(Method::getName)
                .or(() -> extensionContext.getTestClass().map(Class::getSimpleName))
                .orElseThrow();
        this.namespace = extensionContext.getTestClass()
                .map(Class::getSimpleName)
                .map(name -> name.toLowerCase(Locale.ENGLISH) + "-" + Utils.hashStub(testName))
                .orElse("nullClass");
```